src/overwrite.js

```diff
@@ -13,30 +13,28 @@ module.exports = function overwriteCheckFactory(options) {
 };
 
 function checkFilesExist(options) {
-	const client = knox.createClient(options.getCreds());
-
-	return new Promise((resolve, reject) => {
-		client.list({
-			prefix: options.getUploadPath()
-		}, (err, data) => {
-			if (err) {
-				return reject(err);
-			}
 
-			// AWS errors like invalid key or secret are specified in the data
-			// For more see http://docs.aws.amazon.com/AmazonS3/latest/API/ErrorResponses.html
-			if (data.Code) {
-				return reject(new Error(data.Message));
-			}
+	const creds = options.getCreds();
+	const client = new AWS.S3({
+		accessKeyId: creds.key,
+		apiVersion: '2006-03-01',
+		secretAccessKey: creds.secret
+	});
 
+	const params = {
+		Bucket: creds.bucket,
```

Comment:
this PR does not introduce it, but it feels wrong to return the bucket in the 'creds' object.  a separate method on options makes more sense IMO